Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -0,0 +1,598 @@
+      TrExpr(arg, wr, false);
+    }
+
+    protected override void EmitReturn(List<Formal> outParams, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree CreateLabeledCode(string label, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitBreak(string label, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitYield(ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitAbsurd(string message, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitHalt(IToken tok, Expression messageExpr, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree EmitForStmt(IToken tok, IVariable loopIndex, bool goingUp, string endVarName,
+      List<Statement> body,
+      ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree CreateForLoop(string indexVar, string bound, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree CreateDoublingForLoop(string indexVar, int start, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitIncrementVar(string varName, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitDecrementVar(string varName, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override string GetQuantifierName(string bvType) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree CreateForeachLoop(string tmpVarName, Type collectionElementType,
+      string boundVarName,
+      Type boundVarType, bool introduceBoundVar, IToken tok, out ConcreteSyntaxTree collectionWriter,
+      ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override ConcreteSyntaxTree CreateForeachIngredientLoop(string boundVarName, int L, string tupleTypeArgs,
+      out ConcreteSyntaxTree collectionWriter, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitNew(Type type, IToken tok, CallStmt initCall, ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitNewArray(Type elmtType, IToken tok, List<Expression> dimensions, bool mustInitialize,
+      ConcreteSyntaxTree wr) {
+      throw new NotImplementedException();
+    }
+
+    protected override void EmitLiteralExpr(ConcreteSyntaxTree wr, LiteralExpr e) {
+      if (e.Value is bool value) {
+        wr.Write(value ? "True" : "False");
```

Comment:
This should actually be lowercase `true`/`false` to match the other compilers.

Reply:
Just to be sure, Isn't that depended on how the target language considers the values of `true`/`false`? I think every other language has them in all lowercase but python uses `True`/`False`.

Reply:
Sure, but this compiler should be using Python to implement Dafny semantics, and in Dafny the result of `print true` should be `true`, which is what the other compilers are doing (except for C++ but that's a bug - https://github.com/dafny-lang/dafny/issues/1610 :)

Reply:
Why do you mention printing @robin-aws ? This is about creating a boolean literal in Python, which requires using an upper-case, right?

Reply:
Ah you're right, I was confounding this with the implementation of `print`.

Since we're translating a Dafny `print` statement directly to a Python `print` statement, we don't currently have the opportunity to output `true`/`false` as I stated above. I'm fine with that inconsistency in the name of getting this merged sooner though. The proper solution to that will be to create a small runtime Python file to define helper methods like that, similar to the other languages.